### PR TITLE
Demonstrate how to use a development cluster

### PR DESCRIPTION
**Pull request by pietern**
Note: this requires https://github.com/databricks/cli/pull/1698.

### CODE REVIEW

**Comment by pietern** [comment]
I found that without the TF change, it does wait for the cluster to start on creation but does not wait for the cluster to (re-)start on updates.

**Comment by andrewnester** [comment]
Waiting for the doc link to be available to update in readme

File: knowledge_base/development_cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Development cluster
+
+This example demonstrates how to define and use a development (all-purpose) cluster in a Databricks Asset Bundle.
+
+This bundle defines an `example_job` which is run on a job cluster in production mode.
+
+For the development mode (default `dev` target) the job is overriden to use a development cluster which is provisioned 
+as part of the bundle deployment as well.
+
+For more information, please refer to the [documentation](TBD).
```
Comment (pietern) [inline knowledge_base/development_cluster/README.md:10]: Is there a doc link in the works?

**Review by pietern — COMMENTED** [review]
Feel free to merge when ready (I can't approve because I opened the PR).

File: knowledge_base/development_cluster/README.md (hunk 1 above)
Comment (andrewnester) [reply knowledge_base/development_cluster/README.md:10]: In the works, but not yet published

**Pull request merged**